infra/azure-peerpods/main.tf

```diff
@@ -127,65 +119,45 @@ resource "local_file" "kubeconfig" {
   content    = azurerm_kubernetes_cluster.cluster.kube_config_raw
 }
 
-resource "local_file" "workload_identity" {
-  filename        = "./workload-identity.yaml"
+data "local_file" "id_rsa" {
+  filename = "id_rsa.pub"
+}
+
+resource "local_file" "peer-pods-config" {
```

Comment:
Nothing that needs to be done now, but if we still want to have a "working peerpods cluster" through a single `terraform apply`, using the Kubernetes Terraform provider might be cleaner here.